LXC seems fail to build with systemd

```
make[4]: Entering directory '/home/pashev/tmp/nix-build-lxc-1.1.3.drv-1/lxc-lxc-1.1.3-src/config/init/systemd'
make[5]: Entering directory '/home/pashev/tmp/nix-build-lxc-1.1.3.drv-1/lxc-lxc-1.1.3-src/config/init/systemd'
 /nix/store/qawmaicxcljwd25gsq9dqbw7c3q7hjmi-coreutils-8.24/bin/mkdir -p '/nix/store/a8dkxar1lh5xr8n8vfrj6fpzykxcbja0-lxc-1.1.3/libexec/lxc'
 /nix/store/qawmaicxcljwd25gsq9dqbw7c3q7hjmi-coreutils-8.24/bin/install -c lxc-devsetup lxc-apparmor-load '/nix/store/a8dkxar1lh5xr8n8vfrj6fpzykxcbja0-lxc-1.1.3/libexec/lxc'
/nix/store/qawmaicxcljwd25gsq9dqbw7c3q7hjmi-coreutils-8.24/bin/mkdir -p /nix/store/nmvg33m7xsd687wh6figf5hgiw1bj2qc-systemd-217/lib/systemd/system
/nix/store/qawmaicxcljwd25gsq9dqbw7c3q7hjmi-coreutils-8.24/bin/mkdir: cannot create directory '/nix/store/nmvg33m7xsd687wh6figf5hgiw1bj2qc-systemd-217/lib/systemd/system': Permission denied
Makefile:561: recipe for target 'install-systemd' failed
make[5]: *** [install-systemd] Error 1
```

UPDATE:
I'm running Nix store on Debian. LXC's configure detects Debian, thinks it uses systemd (it does) and tries to install systemd files... into Nix's systemd.

--- a/pkgs/os-specific/linux/lxc/default.nix
+++ b/pkgs/os-specific/linux/lxc/default.nix
@@ -36,6 +36,8 @@ stdenv.mkDerivation rec {
     "--sysconfdir=/etc"
     "--enable-doc"
     "--disable-api-docs"
+    "--with-init-script=none"
+    "--with-distro=nixos" # just to be sure it is "unknown"
   ] ++ optional (libapparmor != null) "--enable-apparmor"
     ++ optional (libselinux != null) "--enable-selinux"
     ++ optional (libseccomp != null) "--enable-seccomp"

